server/src/main/java/io/crate/execution/support/RetryRunnable.java

```diff
@@ -60,12 +65,32 @@ public void run() {
                 if (LOGGER.isDebugEnabled()) {
                     LOGGER.debug("Received RejectedExecutionException, will retry again in {}ms", currentDelay);
                 }
-                scheduler.schedule(retryCommand, currentDelay, TimeUnit.MILLISECONDS);
+                cancellable = threadPool.scheduleUnlessShuttingDown(
```

Comment:
Doesn't make a difference here, just as fyi in case you're not aware: Could also check `isExecutorShutdown` on the `EsRejectedExecutionException`